engines/ultima/ultima8/gumps/minimap_gump.cpp

```diff
@@ -253,23 +257,85 @@ uint32 MiniMapGump::sampleAtPoint(const Item *item, int x, int y)
 	}
 
 	if (c > 0) {
-		return _minimap.format.RGBToColor(RenderSurface::_gamma10toGamma22[r / c], RenderSurface::_gamma10toGamma22[g / c], RenderSurface::_gamma10toGamma22[b / c]);
+		return _format.RGBToColor(RenderSurface::_gamma10toGamma22[r / c], RenderSurface::_gamma10toGamma22[g / c], RenderSurface::_gamma10toGamma22[b / c]);
 	}
 
 	return 0;
 }
 
 void MiniMapGump::saveData(Common::WriteStream *ws) {
 	Gump::saveData(ws);
+
+	// Serialize the PixelFormat
+	ws->writeByte(_format.bytesPerPixel);
+	ws->writeByte(_format.rLoss);
+	ws->writeByte(_format.gLoss);
+	ws->writeByte(_format.bLoss);
+	ws->writeByte(_format.aLoss);
+	ws->writeByte(_format.rShift);
+	ws->writeByte(_format.gShift);
+	ws->writeByte(_format.bShift);
+	ws->writeByte(_format.aShift);
+
+	ws->writeUint32LE(static_cast<uint32>(_minimaps.size()));
+	for (unsigned int i = 0; i < _minimaps.size(); ++i) {
+		const Graphics::Surface &minimap = _minimaps[i];
+		ws->writeUint16LE(minimap.w);
+		ws->writeUint16LE(minimap.h);
+		for (int y = 0; y < minimap.h; ++y) {
+			const uint16 *pixels = (const uint16 *)minimap.getBasePtr(0, y);
+			for (int x = 0; x < minimap.w; ++x) {
+				ws->writeUint16LE(*pixels++);
```

Comment:
I guess this could cause endian-ness problems if the pixel format isn't 16-bit, but not sure of a better way to do it without writing a lot of code for each possible bit-depth.

Since we're also writing pixel format it should be future proof for now, so maybe just an assertion so that any future change will quickly notice they broke it?

Reply:
Asserting the bit-depth is a good move. I don't think we'll want to change it, but if we did it could be handled at that point.
Thumbnail saving supports 4bpp and 2bpp without too much extra handling.